pinot-controller/src/main/java/org/apache/pinot/controller/util/TableMetadataReader.java

```diff
@@ -86,6 +86,16 @@ public JsonNode getSegmentsMetadata(String tableNameWithType, List<String> colum
     return getSegmentsMetadataInternal(tableNameWithType, columns, segmentsToInclude, timeoutMs);
   }
 
+  /**
+   * This method takes in table name and checks if there exists any mismtach in the segments/table configs
+   */
+  public JsonNode getSegmentsReloadCheckMetadata(String tableNameWithType, int timeoutMs)
```

Comment:
Suggest changing the signature to:
`boolean isReloadNeeded(...)`

To reduce the overhead, we can extract this part from `getServerCheckSegmentsReloadMetadata()` into a helper method:
```
    TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
    List<String> serverInstances = _pinotHelixResourceManager.getServerInstancesForTable(tableNameWithType, tableType);
    Set<String> serverInstanceSet = new HashSet<>(serverInstances);
    BiMap<String, String> endpoints = _pinotHelixResourceManager.getDataInstanceAdminEndpoints(serverInstanceSet);
    ServerSegmentMetadataReader serverSegmentMetadataReader =
        new ServerSegmentMetadataReader(_executor, _connectionManager);
    List<String> segmentsMetadata =
        serverSegmentMetadataReader.getCheckReloadSegmentsFromServer(tableNameWithType, serverInstanceSet, endpoints,
            timeoutMs);
```
Then process each server response without constructing a map.

Reply:
I am planning to directly return List<String> segmentsMetadata =
        serverSegmentMetadataReader.getCheckReloadSegmentsFromServer(tableNameWithType, serverInstanceSet, endpoints,
            timeoutMs) instead of just needReload flag. This can be used our code to fetch reload flag and later it can extend to include details of segments to be reloaded corresponding to each server as well.

Reply:
Done. Changed the response so as to extend the functionality later accordingly instead of just returning boolean flag